component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -445,4 +486,183 @@ public String toString() {
+        articleImages.put(key, new ArrayList<ImageUrlDim>());
+      }
+      articleImages.get(key).add(getImgUrlExternal(imgUrl));
+    }
+    List<ImageUrlDim> sortedImages = new ArrayList<>();
+    for (Long imgArea : articleImages.keySet()) {
+      // Image size not extractable is in key == '-1' Don't include too small images.
+      if ((imgArea == -1) || (imgArea >= (MIN_SOCIAL_MEDIA_AREA_SIZE))) {
+        Collections.reverse(articleImages.get(imgArea));
+        sortedImages.addAll(articleImages.get(imgArea));
+      }
+    }
+    return sortedImages;
+  }
+
+  ImageUrlDim getImgUrlExternal(String imgUrl) {
+    if (!imgUrl.startsWith("http://") && !imgUrl.startsWith("https://")) {
+      String action = imgUrl.replaceAll("^.*?/(.*?)/.*$", "$1");
+      String space = imgUrl.replaceAll("^(.*?/){2}(.*?)/.*$", "$2");
+      String docname = imgUrl.replaceAll("^(.*?/){3}(.*?)/.*$", "$2");
+      String filename = imgUrl.replaceAll("^(.*?/){4}(.*?)(|\\?.*)$", "$2");
+      String query = imgUrl.replaceAll("^.*\\?(.*)$", "$1");
+      imgUrl = getContext().getURLFactory().createAttachmentURL(filename, space, docname, action,
+          query, getContext().getDatabase(), getContext()).toString();
+    }
+    return new ImageUrlDim(imgUrl, parseImgUrlDimension(imgUrl, "celwidth"), parseImgUrlDimension(
+        imgUrl, "celheight"));
+  }
+
+  Long getImgUrlSizeKey(String imgUrl) {
+    int w = parseImgUrlDimension(imgUrl, "celwidth");
+    int h = parseImgUrlDimension(imgUrl, "celheight");
+    Long area = new Long(h * w);
+    if (area < 0) {
+      area = area * area;
+    } else if ((h == -1) && (w == -1)) {
+      area = -1l;
+    }
+    return area;
+  }
+
+  int parseImgUrlDimension(String imgUrl, String dimension) {
+    String str = imgUrl.replaceAll("^.*" + dimension + "=(\\d+)(&.*)?$", "$1");
+    try {
+      return Integer.parseInt(str);
+    } catch (NumberFormatException nfe) {
+      LOGGER.debug("Exception while parsing Integer from [{}]", str, nfe);
+    }
+    return -1;
+  }
+
+  public String getExternalUrl() {
+    String fullName = getModelUtils().serializeRefLocal(articleDocRef);
+    try {
+      return context.getWiki().getExternalURL(fullName, "view", "ref=socialmedia", context);
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception while trying to get externalURL for doc {}", articleDocRef, xwe);
+    }
+    return context.getWiki().getURL(articleDocRef, "view", context);
+  }
+
+  /**
+   * Adds social media meta tags to the collector, respecting the configuration. If the tags are
+   * activate the calculated OpenGraph tags are added. Furthermore if a Twitter account is
+   * configured (e.g. @example), the Twitter tags are added. The Twitter card type is configurable,
+   * defaulting to "summary"
+   *
+   * @param language
+   */
+  public void addArticleSocialMediaTagsToCollector(String language) {
+    if (1 == getConfigurationSource().getProperty(BLOG_ARTICLE_SOCIAL_MEDIA_CONF_NAME, 0)) {
```

Comment:
Example? ... Does the ConfigurationSource convert an int saved in a boolean field on the DB to type boolean?